tests/PHPStan/Rules/Functions/data/bug-12317.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types = 1);
+
+namespace Bug12317;
+
+class Uuid {
+	public function __construct(public string $uuid) {}
+	public function __toString() { return $this->uuid; }
+}
+
+class HelloWorld
+{
+	/**
+	 * @param list<Uuid> $a
+	 *
+	 * @return list<string>
+	 */
+	public function sayHello(array $a): array
+	{
+		$b = array_map(
+			array: $a,
+			callback: static fn(Uuid $c): string => (string) $c,
+		);
```

Comment:
please add another test like:

```
$b = array_map(
			callback: static fn(Uuid $c): string => (string) $c,
			array: $a,
		);
```

(parameters in different order, which is allowed for named arguments).

Reply:
I added some more test cases.